Features/confirmed preferences #155196338

fddayan opened the pull request:
https://www.pivotaltracker.com/story/show/155196338

fddayan committed:
[#154747783] Prop mapping for LeaseUpPage

fddayan committed:
[#154747783] Fields extracted

fddayan committed:
[#154747783] Props map added for ListingPage

fddayan committed:
[#154747783] Test files renamed

fddayan committed:
[#154747783] Prop mapping setup for ListingsPage

fddayan committed:
[#154747783] Test added for ListingApplicationsPage

fddayan committed:
[#154747783] Introducing prop mapper for ListingApplicationsPage

fddayan committed:
[#154747783] Prop mapper added for ApplicationEditPage

fddayan committed:
[#154747783] Prop mapper added for FlaggedApplicationsShowPage

fddayan committed:
[#154747783] Prop mapper added for FlaggedApplicationsIndexPage

fddayan committed:
[#154747783] Prop mapper added for ApplicationNewPage

fddayan committed:
[#154747783] Cleanup ListingDetailsContentCard

fddayan committed:
[#154747783] Cleanup ListingDetailsContentCard

fddayan committed:
[#154747783] Cleanup ListingDetailsContentCard

fddayan committed:
[#154747783] Split files ListingDetailsContentCard

fddayan committed:
[#154747783] Cleanup ListingDetailsContentCard

fddayan committed:
[#154747783] Refactor fields

fddayan committed:
[#154747783] Cleanup

fddayan committed:
[#154747783] Mappings added for ListingDetails WIP

fddayan committed:
[#154747783] spec passing for listintdetails

fddayan committed:
[#154747783] Refactor ListingDetailsTable

fddayan committed:
[#154747783] mapProps for ListingPage stable

fddayan committed:
[#154747783] adding spec for ListingPage

fddayan committed:
[#154747783] moved fields to ListingPage

fddayan committed:
[#154747783] mapped field for ListingApplicationPage

fddayan committed:
[#154747783] ListingApplicationsPage mappings

fddayan committed:
[#154747783] Prop Mapping for FlaggedApplicationsIndexPage

fddayan committed:
[#154747783] Cleanup PaperForm component

fddayan committed:
Merge branch 'master' into features/central-mapping-#154747783

fddayan committed:
[#154747783] fieldMappers replaced for soqlToApiMappers

fddayan committed:
[#154747783] Renamed propMapper

fddayan committed:
[#154747783] Renamed sqolToApiMappers

fddayan committed:
[#154747783] Fixed UI errors for Flagged URL

fddayan committed:
[#154747783] Adding Fixture for FlaggedApplicationsShowPage.test.js

fddayan committed:
[#154747783] PropMappers for FlaggedApplicationsShowPage

fddayan committed:
[#154747783] Cleanup

fddayan committed:
[#154747783] Added soql to domain form application mapper

fddayan committed:
[#154747783] Cleaup ApplicationDetails

fddayan committed:
[#154747783] Prop mappings ApplicationPage

fddayan committed:
[#154747783] Cleanup ApplicationPage

fddayan committed:
[#154747783] Supplemental application specs improved

fddayan committed:
[#154747783] Cleanup in supplemental applications

fddayan committed:
[#154747783] Test Supplemental application action

fddayan committed:
[#154747783] Cleanup

fddayan committed:
[#154747783] Moved each mapper to its own file to make up entitites

fddayan committed:
[#154747783] Using mapper in Lease up

fddayan committed:
[#154747783] Mappings for Application Listing

fddayan committed:
[#154747783] Cleanup Flagged Index and Show page

fddayan committed:
[#157663604] Lease up table update status fixed

fddayan committed:
SAVEPOINT

fddayan committed:
[#154747783] Using fields in Applicatio new and edit

fddayan committed:
[#154747783] Mapping missing Applicant_Member on preferences

fddayan committed:
[#154747783] Cleanup mappper names

fddayan committed:
[#154747783] Layout form for confirmed units

fddayan committed:
[#156205423] Loading yes no data

callaghanc committed:
[#154747783] Does some trivial utils cleanup

fddayan committed:
[#154747783] Merged with master

fddayan committed:
Merge branch 'features/central-mapping-#154747783' of github.com:Exygy/sf-dahlia-lap into features/central-mapping-#154747783

fddayan committed:
[#154747783] Cleanups

fddayan committed:
[#154747783] Minor cleanup

fddayan committed:
[#156205423] Cleanup service fields and ada properties

fddayan committed:
Merge branch 'features/central-mapping-#154747783' into features/confirmed-units-#156205423

fddayan committed:
[#156205423] Ada preference check list

fddayan committed:
[#156205423] Domain to Api extracted out to own files

fddayan committed:
[#158477512] Reserved senior saved in api

fddayan committed:
[#156205423] Saving ADA properties and cleanups

fddayan committed:
[#156205423] Cleanup and stable

fddayan committed:
[#156205423] Cleanup

fddayan committed:
[#154747783] Cleanup

fddayan committed:
[#154747783] Cleanup

fddayan committed:
Merge branch 'features/central-mapping-#154747783' into features/confirmed-units-#156205423

fddayan committed:
Saving progress

fddayan committed:
[#154747783] Preferences section fixed

fddayan committed:
Merge branch 'features/central-mapping-#154747783' into features/confirmed-units-#156205423

fddayan committed:
[#158821830] Rendering properties

fddayan committed:
[#158821830] Cleanup. All tests passing

fddayan committed:
[#158821830] Cleanup

fddayan committed:
[#156205423] Merged with master

fddayan committed:
[#156205423] Merged with master

fddayan committed:
[#156205423] Cleanups

fddayan committed:
Merge branch 'features/confirmed-units-#156205423' into chores/application-edit-fields-update-#158821830

fddayan committed:
[#155196338] Adding expandable table with preferences

fddayan committed:
[#158821830] Using new mappers

fddayan committed:
[#158821830] Fixed ApplicationPage mege problems

fddayan committed:
[#158821830] Merged with master

fddayan committed:
[#155196338] Icons based on status

fddayan committed:
[#155196338] Render right fiels in columns

fddayan committed:
[#156205423] fields.yml removed

fddayan committed:
[#155196338] Render right icon

fddayan committed:
[#156205423] Merged with master

fddayan committed:
[#155196338] Showing name of invidual preference for combined preference

fddayan committed:
[#155196338] Snapshot updated

fddayan committed:
[#155196338] Rendering proof files

fddayan committed:
[#155196338] Rendering Attachment list

fddayan committed:
[#155196338] Merged with master

fddayan committed:
[#155196338] Merged with master

fddayan committed:
[#155196338] Cleanup

fddayan committed:
[#155196338] Cleanups

fddayan committed:
[#158821830] Organization_URL removed

fddayan committed:
[#158821830] Removed fields.yml

fddayan committed:
[#158821830] Typo fixed

fddayan committed:
Merge branch 'chores/application-edit-fields-update-#158821830' into features/confirmed-preferences-#155196338

fddayan committed:
[#158821830] Tookit updated

fddayan committed:
Merge branch 'chores/application-edit-fields-update-#158821830' into features/confirmed-preferences-#155196338

fddayan committed:
[#158821830] Loading preferences on mount

fddayan committed:
[#158821830] Typo field

fddayan committed:
Merge branch 'chores/application-edit-fields-update-#158821830' into features/confirmed-preferences-#155196338

fddayan committed:
[#155196338] Wrapping table in tookit styles

fddayan committed:
[#158821830] Comment removed

fddayan committed:
Merge branch 'chores/application-edit-fields-update-#158821830' into features/confirmed-preferences-#155196338

callaghanc committed:
[#155196338] Uses document type to label app pref attachments

fddayan committed:
[#155196338] Merged with master

callaghanc commented on app/javascript/components/mappers/soqlToDomain/application_preference.js:
```diff
@@ -14,12 +14,14 @@ export const mapApplicationPreference = (value) => {
+    lw_type_of_proof: value.LW_Type_of_Proof,
     opt_out:value.Opt_Out,
     lottery_status:value.Lottery_Status,
+    post_lottery_validation:value.Post_Lottery_Validation,
```
@fddayan our style so far has been to put spaces after colons, would you have any objection to me reformatting this file to use that convention?

fddayan replied:
not at all,  go for it

callaghanc committed:
[#155196338] Updates several aspects of Confirmed Preferences display

This commit: 1) Adds the ability for the ExpandableTable component to have classes applied to its cells. 2) Right-aligns the Preference Rank column content in the Confirmed Preferences table. 3) Makes it so that if a preference in the Confirmed Preferences table has Invalid status, it gets a red "x" icon; if it has Confirmed status, it gets a green check icon; and if it has any status other than Invalid or Confirmed, it gets no icon. 4) Also does some minor formatting cleanup

callaghanc commented:
@fddayan this is almost good to go, we need to update the rules about which preferences are shown in the confirmed preferences table, i'll push a commit for that soon

callaghanc committed:
[#155196338] Updates rule used to select which prefs to display as confirmed prefs

callaghanc committed:
[#155196338] Adds CSS class to improve display of confirmed prefs proof links

callaghanc commented:
@fddayan Ok I updated the rule used to select which prefs to display in the Confirmed Preferences section. I'm going to add an approving review to this PR, but I would also like you to review the commits I added and add your approving review to this PR as well before we deliver it. Also, you will need to update the review instructions on the related story. I suggest having two applications, so that you can show the alternate names of the prefs that have two name possibilities.

callaghanc commented:
@fddayan I guess you can't technically add an actual review to this PR since it's yours, so just look over the changes I added and comment here your approval/request for changes.

fddayan commented:
@callaghanc I reviewed the commits, look good
